actors/miner/src/lib.rs

```diff
@@ -1728,6 +1728,96 @@ impl Actor {
         Ok(())
     }
 
+    fn prove_commit2(
+        rt: &impl Runtime,
+        params: ProveCommit2Params,
+    ) -> Result<(), ActorError> {
+        // XXX should requireActivationSuccess apply to precommit validation / proof correctness ?
+        // Or just data activation?
+        let state: State = rt.state()?;
+        let info = get_miner_info(rt.store(), &state)?;
+        rt.validate_immediate_caller_is(
+            info.control_addresses.iter().chain(&[info.worker, info.owner]),
+        )?;
+        let store = rt.store();
+
+        let sector_numbers = params.sector_activations.iter().map(|sa| sa.sector_number).collect();
+        let precommits =
+            state.get_all_precommitted_sectors(store, sector_numbers).map_err(|e| {
+                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to get precommits")
+            })?;
+
+        if precommits.is_empty() {
+            return Err(actor_error!(
+                illegal_argument,
+                "zero precommits read from state"
+            ));
+        }
+
+        assert_eq!(len(params.sector_activations), len(params.proofs));
+        let proof_lengths = params.proofs.iter().map(|p| len(p)).collect();
+
+        let (batch_return, proof_infos) = validate_precommits(rt, &precommits, &proof_lengths, true)?;
+        if batch_return.success_count == 0 {
+            return Err(actor_error!(
+                illegal_argument,
+                "no valid precommits specified"
+            ));
+        }
+        let valid_proof_infos = batch_return.successes(&proof_infos);
```

Comment:
Eligible rather than valid? They haven't been proven/validated yet (the precommits have).